packages/edit-site/src/components/home-view-preview/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * WordPress dependencies
+ */
+
+import { store as coreStore } from '@wordpress/core-data';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+
+import Editor from '../editor';
+
+export function HomeViewPreview() {
+	const { isBlockBasedTheme, siteUrl } = useSelect( ( select ) => {
+		const { getEntityRecord, getCurrentTheme } = select( coreStore );
+		const siteData = getEntityRecord( 'root', '__unstableBase' );
+
+		return {
+			isBlockBasedTheme: getCurrentTheme()?.is_block_theme,
+			siteUrl: siteData?.home,
+		};
+	}, [] );
+
+	// If theme is block based, return the Editor, otherwise return the site preview.
+	return isBlockBasedTheme ? (
+		<Editor />
+	) : (
+		<iframe
+			src={ siteUrl }
+			title="front-end view"
```

Comment:
```suggestion
			title={ __( 'Site Preview' ) }
```

It might be a good idea to use the same text as [the iframe title in the customizer](https://github.com/t-hamano/wordpress-develop/blob/a27e6a8ecd5279a45e47ef88d048643740d07b5a/src/wp-includes/script-loader.php#L1298).